jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessors.java

```diff
@@ -65,4 +80,40 @@ private static String determineJarType(Path jarPath) throws IOException {
       return STANDARD;
     }
   }
+
+  /**
+   * Determines the java version of JAR. Derives the version from the first .class file it finds in
+   * the JAR.
+   *
+   * @param jarPath path to the jar
+   * @return java version
+   * @throws IOException if I/O exception thrown when opening the jar file
+   */
+  private static Integer getVersion(Path jarPath) throws IOException {
+    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
+      Enumeration<JarEntry> jarEntries = jarFile.entries();
+      while (jarEntries.hasMoreElements()) {
+        String jarEntry = jarEntries.nextElement().toString();
+        if (jarEntry.endsWith(".class")) {
+          URLClassLoader loader = new URLClassLoader(new URL[] {jarPath.toUri().toURL()});
+          try (InputStream classFileInfo = loader.getResourceAsStream(jarEntry)) {
+            DataInputStream data = new DataInputStream(classFileInfo);
```

Comment:
Can we do this?

```suggestion
          try (InputStream classFileInfo = loader.getResourceAsStream(jarEntry);
              DataInputStream data = new DataInputStream(classFileInfo)) {
```
And why is it called classFile*Info*? Isn't it really a class file?

Or, in most cases, I believe this is OK:
```java
try (InputStream classFile = new DataInputStream(loader.getResourceAsStream(...))) {
```